rskj-core/src/main/java/org/ethereum/core/BlockHeader.java

```diff
@@ -471,7 +498,30 @@ public String getParentShortHash() {
         return HashUtil.shortHash(getParentHash().getBytes());
     }
 
-    private static BigInteger parseBigInteger(byte[] bytes) {
-        return bytes == null ? BigInteger.ZERO : BigIntegers.fromUnsignedByteArray(bytes);
+    public byte[] getMiningForkDetectionData() {
+        if(includeForkDetectionData) {
+            if (hasMiningFields()) {
+                byte[] encodedBlock = getEncoded(false, false);
+                byte[] hashForMergedMining = HashUtil.keccak256(encodedBlock);
+
+                byte[] hashForMergedMiningPrefix = Arrays.copyOfRange(hashForMergedMining, 0, HASH_FOR_MERGED_MINING_PREFIX_LENGTH);
+                byte[] coinbaseTransaction = getBitcoinMergedMiningCoinbaseTransaction();
+
+                List<Byte> hashForMergedMiningPrefixAsList = Arrays.asList(ArrayUtils.toObject(hashForMergedMiningPrefix));
+                List<Byte> coinbaseAsList = Arrays.asList(ArrayUtils.toObject(coinbaseTransaction));
+
+                int position = Collections.lastIndexOfSubList(coinbaseAsList, hashForMergedMiningPrefixAsList);
+                if (position == -1) {
+                    return new byte[12];
+                }
+
+                int from = position + HASH_FOR_MERGED_MINING_PREFIX_LENGTH;
+                return Arrays.copyOfRange(coinbaseTransaction, from, from + FORK_DETECTION_DATA_LENGTH);
```

Comment:
I see many calls to getMiningForkDetectionData() around the code. I would suppose that the first time it is called the miningForkDetectionData is extracted, but the following calls, if the miningForkDetectionData field is non-null, it would be returned (as a cache), instead of being re-computed.
Is there a reason for continuous recomputation?

Reply:
Worst case scenario, it is called only twice in a row. All the other calls are done to different instances. I think twice re computation does not need a cache, specially in a case where the computational complexity is very low.
Anyway, I added a cache as you asked :)